src/vxworks/mod.rs

```diff
@@ -100,43 +96,22 @@ pub type SD_ID = ::OBJ_HANDLE;
 pub type CONDVAR_ID = ::OBJ_HANDLE;
 
 // From vxTypes.h
-pub type _Vx_usr_arg_t = ::ssize_t; // c_int for LP32
-pub type _Vx_exit_code_t = ::ssize_t; // c_int for LP32
+pub type _Vx_usr_arg_t = isize;
+pub type _Vx_exit_code_t = isize;
 pub type _Vx_ticks_t = ::c_uint;
 pub type _Vx_ticks64_t = ::c_ulonglong;
 
-// From vxTypesBase.h
-pub type va_list = *mut ::c_char;
-
 pub type sa_family_t = ::c_uchar;
 
 // structs that only exist in userspace
 s! {
-    // b_struct_vx_eventsResourceCb.h
-    pub struct _Vx_EVENTS_RSRC {
-        pub registered : ::c_uint,
-        pub taskId     : ::c_int,
-        pub options    : ::c_uchar,
-        pub pad        : [::c_uchar; 3],
-    }
-
-    // b_struct_vx_semaphore.h
-    pub struct _Vx_semaphore {
-        pub magic  : ::c_uint,
-        pub semType: ::c_uint,
-        pub options: ::c_uint,
-        pub recurse: ::c_uint,
-        pub owned_k: ::c_uint, // owned_k is volatile
-        pub semId_k: ::_Vx_SEM_ID_KERNEL,
-        pub state  : ::c_uint, //state is union of _Vx_UINT and _Vx_UINT
-        pub events : ::_Vx_EVENTS_RSRC,
-    }
+    pub struct _Vx_semaphore {}
```

Comment:
What's the definition of this C type ?

Reply:
Do we have to add the definition here? We want to hide the internal details for this struct if possible.

Reply:
I don't know. Why are you adding it here ? (that's why I asked for the C definition, if this is a public type that's used on some public API, then it might make sense adding it here, but since C does not have types of zero-size, the looks of this type are quite suspicious).

Reply:
I will remove "pub" for the struct. We only want o make the type of pointer to this struct be public which is defined at line 90:

pub type _Vx_SEM_ID = *mut ::_Vx_semaphore;

Reply:
Removing "pub" would result compiling error.

Reply:
I find a example in hermit/mod.rs which is using the similiary code:
    pub struct utsname {}
...
    pub fn uname(buf: *mut ::utsname) -> ::c_int;

Is such code not recommented any more?

Reply:
Hermit is not a supported platform, in fact it didn't even compile for a long time and nobody noticed. 

The only properly supported platforms are those for which `libc-test` runs and passes correctly on the platform.

Reply:
Running libc-test on your platform should catch all of these issues.

Reply:
@gnzlbg As I mentioned above, we did recently get libc-test working (thanks to your suggestion in an earlier review), and apparently (with the changes under review here) it passes, i.e. it does not complain about this.

As @BaoshanPang mentioned, the intent here is to provide an "opaque" type, i.e. clients of this interface only need to deal with pointers to this type, and should not (nor need to) know the inner workings.

Reply:
However let me double check, because we may be "skipping" some tests.

Reply:
I think we are skipping `_Vx_semaphore` itself (the type we want to keep opaque), but the types used in the APIs that are defined in terms of this type (e.g. _Vx_SEM_ID), pass the libc-test:

```
[172] _Vx_SEM_ID size => Passed
[173] _Vx_SEM_ID align => Passed

```

Reply:
> As @BaoshanPang mentioned, the intent here is to provide an "opaque" type, i.e. clients of this interface only need to deal with pointers to this type, and should not (nor need to) know the inner workings.

The way this is done is by using an enum, e.g., `pub enum _Vx_semaphore{}` - users cannot create a value of that type because it has no variants.

Reply:
Thanks, we'll make that change.